prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -987,6 +987,13 @@ static int efa_rdm_ep_close(struct fid *fid)
 	}
 
 	if (efa_rdm_ep->shm_ep) {
+		ret = fi_close(&efa_rdm_ep->shm_srx->fid);
```

Comment:
We have other places that can close shm resources, including https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rdm/efa_rdm_ep_fiops.c#L1048-L1059

Reply:
And our unit tests have tons of places that replies on closing shm ep hackily as `fi_close`: https://github.com/ofiwg/libfabric/blob/main/prov/efa/test/efa_unit_test_ep.c#L126-L131. These tests are all failed by this change. I think we need to modify our unit tests that doesn't close shm ep explicitly in the test.

Reply:
@shijin-aws Oh that is funky. I didn't realize that. I added a ref count in the shm srx import to make sure it gets closed properly even though there isn't really anything to be closed for shm so I think dropping this close will cause failures in the regular case. I think you probably at minimum need a wrapper around that hack to close both the shm ep and the shm srx. Do you think that would be enough?

Reply:
@shijin-aws I am preparing a patch to fix that. We can use `fi_setopt(.. FI_OPT_SHARED_MEMORY_PERMMITED, false)` to disable shm in the test.

Reply:
Sweet, thank you!

Reply:
But generally it is annoying that provider needs to manage more resources for peer providers

Reply:
I agree, but I think we need to match all the peer resources to the same flow - the peer cq and peer counter are the same (efa needs to close the shm cq and counters). I think it needs to happen and this one was just missed so it's annoying to add it in later. That would be a perk of the link provider, to remove that handling from efa

Reply:
I opened a PR to clean up our tests: https://github.com/ofiwg/libfabric/pull/10406. we need to rebase your PR after https://github.com/ofiwg/libfabric/pull/10406 is merged and then make sure shm resources are cleaned correctly in https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rdm/efa_rdm_ep_fiops.c#L1048-L1059 as well.